`encrypt--jdbc--standalone--local` in the example module throws an npe

The initialization of the storage node was inconsistent at various places, either datasource name or host + user+ port causing an exception to get a datasource.

![d6NyM6ssvo](https://github.com/apache/shardingsphere/assets/29777558/4064d4f0-1e5e-46d4-ad64-810845f35c8d)

**exception**
```
java.sql.SQLException: Unknown exception: null
    at org.apache.shardingsphere.infra.exception.core.external.sql.ShardingSphereSQLException.toSQLException (ShardingSphereSQLException.java:62)
    at org.apache.shardingsphere.infra.exception.dialect.SQLExceptionTransformEngine.toSQLException (SQLExceptionTransformEngine.java:68)
    at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.executeUpdate (ShardingSphereStatement.java:264)
    at org.apache.shardingsphere.example.readwritesplitting.jdbc.repository.OrderRepository.dropTable (OrderRepository.java:[53](https://github.com/apache/shardingsphere/actions/runs/7239568505/job/19721647465#step:10:54))
    at org.apache.shardingsphere.example.readwritesplitting.jdbc.service.ExampleService.cleanEnvironment (ExampleService.java:136)
    at org.apache.shardingsphere.example.readwritesplitting.jdbc.service.ExampleService.run (ExampleService.java:55)
    at org.apache.shardingsphere.example.readwritesplitting.jdbc.ExampleMain.main (ExampleMain.java:32)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:2[54](https://github.com/apache/shardingsphere/actions/runs/7239568505/job/19721647465#step:10:55))
    at java.lang.Thread.run (Thread.java:829)
Caused by: java.lang.NullPointerException
    at org.apache.shardingsphere.infra.datasource.pool.CatalogSwitchableDataSource.getConnection (CatalogSwitchableDataSource.java:46)
    at org.apache.shardingsphere.infra.database.DatabaseTypeEngine.getStorageType (DatabaseTypeEngine.java:118)
    at org.apache.shardingsphere.infra.database.DatabaseTypeEngine.getProtocolType (DatabaseTypeEngine.java:62)
    at org.apache.shardingsphere.metadata.factory.ExternalMetaDataFactory.create (ExternalMetaDataFactory.java:54)
    at org.apache.shardingsphere.mode.manager.context.ConfigurationContextManager.createChangedDatabase (ConfigurationContextManager.java:378)
    at org.apache.shardingsphere.mode.manager.context.ConfigurationContextManager.createChangedDatabases (ConfigurationContextManager.java:349)
    at org.apache.shardingsphere.mode.manager.context.ConfigurationContextManager.createMetaDataContexts (ConfigurationContextManager.java:325)
    at org.apache.shardingsphere.mode.manager.context.ConfigurationContextManager.alterRuleConfiguration (ConfigurationContextManager.java:155)
    at org.apache.shardingsphere.mode.manager.standalone.NewStandaloneModeContextManager.alterRuleConfiguration (NewStandaloneModeContextManager.java:259)
    at org.apache.shardingsphere.infra.connection.refresher.type.table.DropTableStatementSchemaRefresher.refresh (DropTableStatementSchemaRefresher.java:49)
    at org.apache.shardingsphere.infra.connection.refresher.type.table.DropTableStatementSchemaRefresher.refresh (DropTableStatementSchemaRefresher.java:37)
    at org.apache.shardingsphere.infra.connection.refresher.MetaDataRefreshEngine.refresh (MetaDataRefreshEngine.java:69)
    at org.apache.shardingsphere.driver.executor.DriverJDBCExecutor.doExecute (DriverJDBCExecutor.java:149)
    at org.apache.shardingsphere.driver.executor.DriverJDBCExecutor.executeUpdate (DriverJDBCExecutor.java:97)
    at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.useDriverToExecuteUpdate (ShardingSphereStatement.java:371)
    at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.executeUpdate (ShardingSphereStatement.java:320)
    at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.executeUpdate0 (ShardingSphereStatement.java:341)
    at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.executeUpdate (ShardingSphereStatement.java:259)
    at org.apache.shardingsphere.example.readwritesplitting.jdbc.repository.OrderRepository.dropTable (OrderRepository.java:53)
    at org.apache.shardingsphere.example.readwritesplitting.jdbc.service.ExampleService.cleanEnvironment (ExampleService.java:136)
    at org.apache.shardingsphere.example.readwritesplitting.jdbc.service.ExampleService.run (ExampleService.java:[55](https://github.com/apache/shardingsphere/actions/runs/7239568505/job/19721647465#step:10:56))
    at org.apache.shardingsphere.example.readwritesplitting.jdbc.ExampleMain.main (ExampleMain.java:32)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:254)
    at java.lang.Thread.run (Thread.java:[82](https://github.com/apache/shardingsphere/actions/runs/7239568505/job/19721647465#step:10:83)9)
```

--- a/infra/common/src/main/java/org/apache/shardingsphere/infra/config/database/impl/DataSourceProvidedDatabaseConfiguration.java
+++ b/infra/common/src/main/java/org/apache/shardingsphere/infra/config/database/impl/DataSourceProvidedDatabaseConfiguration.java
@@ -69,7 +69,8 @@ public final class DataSourceProvidedDatabaseConfiguration implements DatabaseCo
         for (Entry<String, DataSourcePoolProperties> entry : dataSourcePoolPropsMap.entrySet()) {
             String storageUnitName = entry.getKey();
             StorageNode storageNode = storageUnitNodeMap.get(storageUnitName);
-            StorageUnit storageUnit = new StorageUnit(storageNode, entry.getValue(), storageNodeDataSources.get(storageNode));
+            DataSource dataSource = storageNodeDataSources.containsKey(storageNode) ? storageNodeDataSources.get(storageNode) : storageNodeDataSources.get(new StorageNode(storageUnitName));
+            StorageUnit storageUnit = new StorageUnit(storageNode, entry.getValue(), dataSource);
             result.put(storageUnitName, storageUnit);
         }
         return result;

